trieste/bayesian_optimizer.py

```diff
@@ -788,152 +822,152 @@ def _write_summary_init(
-            for i in tf.range(tf.shape(query_points)[1]):
-                if len(query_points) == 1:
-                    logging.scalar(f"query_points/[{i}]", float(query_points[0, i]))
-                else:
-                    logging.histogram(f"query_points/[{i}]", query_points[:, i])
-            logging.histogram("query_points/euclidean_distances", lambda: pdist(query_points))
-
-        if pd and sns and logging.include_summary("query_points/_pairplot"):
-            columns = [f"x{i}" for i in range(tf.shape(query_points)[1])]
-            qp_preds = query_points
-            for tag in datasets:
-                qp_preds = tf.concat(
-                    [qp_preds, tf.cast(models[tag].predict(query_points)[0], query_points.dtype)], 1
+            if tf.size(datasets[tag].observations) > 0:
+                logging.scalar(
+                    f"{tag}.observation{suffix}/best_overall",
+                    np.min(datasets[tag].observations[..., i]),
                 )
-                columns.append(f"{tag} predicted")
-            query_new_df = pd.DataFrame(qp_preds, columns=columns).applymap(float)
-            query_new_df["query points"] = "new"
-            query_plot_df = pd.concat(
-                (query_plot_dfs.get(0), query_new_df), copy=False, ignore_index=True
+
+        if logging.include_summary(f"{tag}.observations/_pairplot") and (
+            pd and sns and output_dim >= 2
+        ):
+            columns = [f"x{i}" for i in range(output_dim)]
+            observation_new_df = pd.DataFrame(
+                tagged_output[tag].observations, columns=columns
+            ).applymap(float)
+            observation_new_df["observations"] = "new"
+            observation_plot_df = pd.concat(
+                (observation_plot_dfs.get(tag), observation_new_df),
+                copy=False,
+                ignore_index=True,
             )
-            pairplot = sns.pairplot(query_plot_df, hue="query points", height=2.25)
-            padding = 0.025 * (self._search_space.upper - self._search_space.lower)
-            upper_limits = self._search_space.upper + padding
-            lower_limits = self._search_space.lower - padding
-            for i in range(self._search_space.dimension):
-                pairplot.axes[0, i].set_xlim((lower_limits[i], upper_limits[i]))
-                pairplot.axes[i, 0].set_ylim((lower_limits[i], upper_limits[i]))
-            logging.pyplot("query_points/_pairplot", pairplot.fig)
-            query_plot_df["query points"] = "old"
-            query_plot_dfs[0] = query_plot_df
-
-        logging.scalar("wallclock/step", total_step_wallclock_timer.time)
-        logging.scalar(
-            "wallclock/query_point_generation",
-            query_point_generation_timer.time,
-        )
-        logging.scalar(
-            "wallclock/model_fitting",
-            model_fitting_timer.time,
+            observation_plot_df["pareto"] = non_dominated(datasets[tag].observations)[1] == 0
```

Comment:
actually, it only makes sense to look at the Pareto if the tag is OBJECTIVE?

Reply:
actually I thought the EHVI etc. were expected specifically an OBJECTIVE tag but you can specify whatever tag you want as `objective_tag`, so this may be very tricky.

Reply:
(I've changed it so we only highlight pareto fronts for OBJECTIVE tag or for any tag in a single-tag dataset — at least for now)